### ISSUE TITLE
Get Object Metadata Changes: If-Modified-Since header

### ISSUE THREAD

**Issue by huwd** [restforce/restforce#447]
Heya, 

I've been checking out the [Get Object Metatdata Changes](https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/sobject_describe_with_ifmodified_header.htm) docs here. 

Looks like since v29.0 there's been a nifty feature where you can call SObject Describe with a custom header `If-Modified-Since: Wed, 3 Jul 2013 19:43:31 GMT`.

It seems like this will optionally return one of two things:
1. If there have been no changes to the object since the date supplied (July 3rd 2013) then you get 
```
HTTP/1.1 304 Not Modified
Date: Fri, 12 Jul 2013 05:03:24 GMT
```
2. Otherwise if you've got new custom fields made after that date you get the object.describe metatdata as usual.

I've got a stupid roundtrip problem that this could really help with, I'm doing a lot of comparing a remote schema to one I've had to make a record of locally. This could stop me doing a whole bunch of needless comparisons!

I can't find reference to this in the docs (apologies if I've missed it!), am I right in thinking this isn't supported just yet?

If not this is probably worth me forking and having a go at myself, I'd be happy to put in a PR, but I could do with a couple of pointers:

### Custom headers feature?
I can see since I last used restforce there's been some work on adding [custom headers](https://github.com/restforce/restforce/blob/2cdeb26bd78851c9fb8d39f16c115d74e91dfc25/spec/unit/middleware/custom_headers_spec.rb).

The [SF spec](https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/headers.htm) seems to have some pretty specific use cases. And the readme describes how you can use the builder feature to modify _all_ [client requests](https://github.com/restforce/restforce/blob/master/README.md#custom-headers).

I'm not sure if that's what I want to do here, I think I only want to selectively fire these on some describe requests. Ideally I could hand describe an optional prop `describe('MyObject__c', 'Fri, 12 Jul 2013 05:03:24 GMT') and it would ask Faraday to make the connection with a different config...

But as much of this is done in middleware, and the Mash seems fairly independent I'm not sure about where the best point for those to interact might be?

### Making a second connection
If my ideal API of `.describe` is a non starter, or isn't a good idea for other reasons I could just create a second client set up with a different config and only call that when i need the 304 feature...

That might be my quick fix for the app but it feels kinda icky having 2 client connections floating about... What do other Dev here think, i'm either being overly picky or about to step on a hidden landmine?

### Overriding a faraday config on a per connection basis?
Only other option i can think of is putting in a switch that modified the Faraday config on the fly for some connections then resets it after like have a setup and teardown in a single method... but that also seems a bit icky.

Anyone tried using this before?
Am I missing some code already here?
Anyone else have a usecase for this?

If so, I'll put in a PR?

**Comment by matt17r** [comment]
A bit delayed but I'm interested in this. I'm still in the early stages of exploration and I may find some other workaround so I'm also interested to hear if you abandoned the idea in favour of something else.

**Comment by huwd** [comment]
Wow this is a blast from the past. Not thought about this in what seems like a very long time.

I went back and looked at an old codebase, been a while but I suspect I did crack this one.

I have the following method:

```ruby
# get last modified header off all SObjects
  def last_modified_remote
    @restforce.get("/services/data/v#{ENV.fetch('SALESFORCE_API_VERSION')}/sobjects")['Last-Modified']
  end
```

The `@restforce` is an instance of my client.
I then use that to do a comparison with a stored date and conditionally choose if to proceed or not.

**Comment by huwd** [comment]
Might be fair to say I'm unlikely to open that PR now 😅

So @matt17r whilst I'm happy to have a chat about any old source code I have and how it works, this issue can probably be closed down and archived 😁

**Comment by matt17r** [comment]
Yep. I’ll figure something else out. Can you close it (as OP) or does a maintainer have to close it?
